study-builder/studies/pancan/study-events.conf

```diff
@@ -101,15 +101,37 @@
         "type": "ACTIVITY_INSTANCE_CREATION",
         "activityCode": ${id.act.about_cancer},
         "createFromAnswer": true,
-        "sourceQuestionStableId": ${id.q.primary_cancer_list},
+        "sourceQuestionStableId": ${id.q.primary_cancer_list_self},
         "targetQuestionStableId": ${id.q.diagnosis_type}
       },
       # If there's already about-cancer instances, then don't create any more.
-      "preconditionExpr": """!user.studies["cmi-pancan"].forms["ABOUT_CANCER"].hasInstance()""",
+      "preconditionExpr": """!user.studies["cmi-pancan"].forms["ABOUT_CANCER"].hasInstance()
+      && (user.studies["cmi-pancan"].forms["PREQUAL"].questions["DESCRIBE"].answers.hasOption("DIAGNOSED"))""",
       "maxOccurrencesPerUser": 1,
       "dispatchToHousekeeping": false,
       "order": 1
     },
+    {
```

Comment:
FYI I've moved things around in my email PR #903, so there might be some merge conflicts.